apps/faucet/src/components/ProcessingModal.tsx

```diff
@@ -107,6 +109,41 @@ const ProcessingModal = () => {
     }
   }, [mintTokenRes]);
 
+  const errorMessage = useMemo(() => {
+    if (!mintTokenRes) return '';
```

Comment:
I see there are 4 nested if statement right here. Maybe with this approach the code is easier to read (suggested by chatgpt):

```
const getErrorMessage = (mintTokenRes) => {
  if (!mintTokenRes) return '';

  const defaultMessage = 'Oops, the transfer could not be completed.';

  if (!FaucetError.isFaucetError(mintTokenRes)) {
    return defaultMessage;
  }

  const errorCode = mintTokenRes.getErrorCode();

  if (errorCode !== FaucetErrorCode.TOO_MANY_CLAIM_REQUESTS) {
    return defaultMessage;
  }

  const payload = mintTokenRes.getPayload();

  if (!payload || !('lastClaimedDate' in payload)) {
    return defaultMessage;
  }

  const { lastClaimedDate, claimPeriod } = payload;
  const timeLeft = Date.now() - lastClaimedDate.getTime() + claimPeriod;
  const timeStr = formatTimeLeft(timeLeft);

  if (timeStr) {
    return `You have already claimed within the specified time period. Please wait ${timeStr} before making another claim.`;
  }

  return defaultMessage;
};
```